miner/src/pool/ready.rs

```diff
@@ -129,6 +129,38 @@ impl txpool::Ready<VerifiedTransaction> for Condition {
 	}
 }
 
+pub struct OptionalState<C> {
```

Comment:
Can we add some docs here?

"Checks readiness of transactions by comparing the nonce to state nonce. If nonce isn't found in provided state nonce store, defaults to the tx nonce and updates the nonce store. Useful for using with a state nonce cache when false positives are allowed." ?